channeldb/graph.go

```diff
@@ -1158,11 +1158,21 @@ func putLightningNode(nodeBucket *bolt.Bucket, aliasBucket *bolt.Bucket, node *L
 		return err
 	}
 
-	addrString := node.Address.String()
-	if err := wire.WriteVarString(&b, 0, addrString); err != nil {
+	numAddresses := uint16(len(node.Addresses))
+	byteOrder.PutUint16(scratch[:2], numAddresses)
+	if _, err := b.Write(scratch[:2]); err != nil {
 		return err
 	}
 
+	for _, address := range node.Addresses {
+		if err := wire.WriteVarString(&b, 0, address.Network()); err != nil {
```

Comment:
iirc whether an address is IPv4 or IPv6 it'll always return a "tcp" if it's a `net.TCPAddr`. Therefore, instead I think we can just rely on the `String` method of the `net.Addr` itself here. It would also be a bit more compact (as we can leave out the network).